ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -64,13 +72,23 @@
   static final private String CLASS_NAME = Cleaner.class.getName();
   static final private Logger LOG = LoggerFactory.getLogger(CLASS_NAME);
   private long cleanerCheckInterval = 0;
+  private ExecutorService cleanerExecutor;
+  private CompletionService<Void> completionService;
 
   private ReplChangeManager replChangeManager;
 
   @Override
   public void init(AtomicBoolean stop) throws Exception {
     super.init(stop);
     replChangeManager = ReplChangeManager.getInstance(conf);
+    ThreadFactory threadFactory = new ThreadFactoryBuilder()
```

Comment:
Shall we move this to the run method too? It would make it easier to understand the code IMHO

Reply:
Or maybe use this as an utility method?

Reply:
Done.